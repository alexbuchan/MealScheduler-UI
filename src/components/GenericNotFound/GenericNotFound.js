import React from 'react';

const GenericNotFound = ({ location }) => {
  console.log(location.pathname)
  return (
    <div className='generic-not-found'>
    <h1>404: Not Found</h1>
    <h3>Page not found.</h3>
    </div>
  );
}

export default GenericNotFound;