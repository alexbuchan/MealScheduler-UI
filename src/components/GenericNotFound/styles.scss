.generic-not-found {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 200px;
}