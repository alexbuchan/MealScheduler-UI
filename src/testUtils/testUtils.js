import React from 'react';
import { shallow } from 'enzyme';

export const componentSetup = (Component, props = {}, options = {}) => {
  const instance = shallow(<Component { ...props } />, options);
  return instance;
}

export const findByTestAttribute = (instance, attribute) => {
  const wrapper = instance.find(`[data-test="${attribute}"]`);
  return wrapper;
}