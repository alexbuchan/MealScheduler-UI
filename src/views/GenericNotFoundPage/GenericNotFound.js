import React from 'react';

// No props.
// const propTypes = {};

const GenericNotFound = () => {
  return (
    <div className='generic-not-found'>
      <h1>404: Not Found</h1>
      <h3>Page not found.</h3>
    </div>
  );
}

// GenericNotFound.propTypes = propTypes;
export default GenericNotFound;