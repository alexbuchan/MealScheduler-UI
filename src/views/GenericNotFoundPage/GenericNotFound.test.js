import GenericNotFound from './GenericNotFound';
import { componentSetup } from '../../testUtils/testUtils';

describe('GenericNotFound', () => {
  let instance;
  
  beforeEach(() => {
    instance = componentSetup(GenericNotFound);
  });

  it('Should render without errors', () => {
    expect(instance).toMatchSnapshot();
  });
});