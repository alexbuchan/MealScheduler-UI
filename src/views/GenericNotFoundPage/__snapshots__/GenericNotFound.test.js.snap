// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericNotFound Should render without errors 1`] = `
<div
  className="generic-not-found"
>
  <h1>
    404: Not Found
  </h1>
  <h3>
    Page not found.
  </h3>
</div>
`;
